Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -603,6 +617,16 @@ namespace AzToolsFramework
             return false;
         }
 
+        bool PrefabPublicHandler::IsOverrideEditing(const Instance& owningInstance) const
+        {
+            auto editorEntityContextId = AzFramework::EntityContextId::CreateNull();
+            EditorEntityContextRequestBus::BroadcastResult(editorEntityContextId, &EditorEntityContextRequests::GetEditorEntityContextId);
+            auto focusedInstance = m_prefabFocusHandler.GetFocusedPrefabInstance(editorEntityContextId);
```

Comment:
What is the reasoning behind the use of auto here ?

Reply:
imo I tended not to use `auto` unless I am seeing either of these:
1. Long type name like iterator or `AZStd::unordered_map<AZStd::string, AZStd::vector<AZ::EntityId>>`
2. The whole line is so long and not easy to read.

But for these two lines here I would change not to use `auto`. Thanks for pointing this out. I think the reason I used here is I was copying this code from elsewhere in the class to keep some style consistency.

If you check other places in the code we frequently use `auto` for 'InstanceOptionalReference' type. But yeah I agree that we may avoid this in the future and probably create a backlog task to clean this up.

Reply:
Might be worth starting right now and not adding auto's to your code, these kinds of backlog tasks have a tendency to never get prioritized if some work hasn't already been done